### PR TITLE
[Algo] : Quick sort implementation

### PR DESCRIPTION
**Pull request by MicoDan**
This is the implementation of the quick sort algorithm in the kin lang programming language

### CODE REVIEW

**Review by pacifiquem — CHANGES_REQUESTED** [review]
Change file name to `quick.kin` to maintain the early format for this repo

File: src/sorting/quicksort
Comment (pacifiquem) [inline src/sorting/quicksort:1]: When tested on these dataset: `[45, 11, 56, 2, 0, 12, 155, 3, 3, 4, 566]`
This was the output:
```txt
45
11
56
2
0
12
155
3
3
4
566
```

**Review by pacifiquem — CHANGES_REQUESTED** [review]

**Comment by MicoDan** [comment]
I don't know if it's an error on your end but the algorithm works perfect for me locally, even testing with that same dataset,
test
 take a look:

**Comment by pacifiquem** [comment]
Not sure what's wrong with kin-lang/kin#117 this PR is good to be merged.

**Review by pacifiquem — APPROVED** [review]

**Comment by pacifiquem** [comment]
This PR will remain open until further testcases on kin-lang/kin#117 are finished. Thanks @MicoDan

**Review by pacifiquem — APPROVED** [review]

**Comment by pacifiquem** [comment]
With issue (kin-lang/kin#119) fixed in [v0.3.1-beta](https://github.com/kin-lang/kin/releases/tag/v0.3.1-beta)

This PR passes all checks, thanks for contributions @MicoDan .

**Pull request merged**